types/dicer/index.d.ts

```diff
@@ -0,0 +1,167 @@
+// Type definitions for dicer 0.2
+// Project: https://github.com/mscdex/dicer
+// Definitions by: Robin Goupil <https://github.com/GoupilRobin>
+//                 BendingBender <https://github.com/BendingBender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+/// <reference types="node" />
+
+export = Dicer;
+
+import stream = require("stream");
+
+// tslint:disable:unified-signatures
+
+/**
+ * A very fast streaming multipart parser for node.js.
+ * Dicer is a WritableStream
+ *
+ * Dicer (special) events:
+ * - on('finish', ()) - Emitted when all parts have been parsed and the Dicer instance has been ended.
+ * - on('part', (stream: PartStream)) - Emitted when a new part has been found.
+ * - on('preamble', (stream: PartStream)) - Emitted for preamble if you should happen to need it (can usually be ignored).
+ * - on('trailer', (data: Buffer)) - Emitted when trailing data was found after the terminating boundary (as with the preamble, this can usually be ignored too).
+ */
+declare class Dicer extends stream.Writable {
+    /**
+     * Creates and returns a new Dicer instance with the following valid config settings:
+     *
+     * @param config The configuration to use
+     */
+    constructor(config: Dicer.Config);
+    /**
+     * Sets the boundary to use for parsing and performs some initialization needed for parsing.
+     * You should only need to use this if you set headerFirst to true in the constructor and are parsing the boundary from the preamble header.
+     *
+     * @param boundary The boundary to use
+     */
+    setBoundary(boundary: string): void;
+    addListener(event: string, listener: (...args: any[]) => void): this;
```

Comment:
The most general signature should go last; otherwise, the type will always resolve to this one.